citar.el

```diff
@@ -1017,6 +969,7 @@ FORMAT-STRING."
   "Look up key for a citar-reference TYPE and TARGET."
   (cons type (or (cadr (assoc target
                               (with-current-buffer (embark--target-buffer)
+                                ;; FIX how?
```

Comment:
@aikrahguzar I think you wrote this, but I don't understand it, and how to adapt.

Reply:
I am pretty sure I didn't write this code but it is looking up the key given the candidate string. So I think the adapted version would something like
```emacs-lisp
(defun citar--reference-transformer (type candidate)
  "Look up key for a citar-reference TYPE and CANDIDATE."
  (with-current-buffer (embark--target-buffer) (cons type (citar-lookup-key-for-candidate candidate))))
```
I don't know enough about the changes here to know exactly what should be `citar-lookup-key-for-candidate` probably a lookup in one of the hashes.

Reply:
> I am pretty sure I didn't write this code  ...

OK; sorry for the distraction :-)

 @roshanshariff  may know then.

It would help to clarify the docstring here. Is this for the minibuffer, or the buffer?

I was assuming the latter, but I guess it doesn/t matter. In either case, the code should just use the key, which we already have.

Reply:
Actually, the problem is specific to `citar-open-entry`, when accessed from `embark-act`.

Reply:
> With both this and your example, I get a "key 'nil' not found" error, which hints to me at a `gethash` not returning a result. 

Is there error in `citar--reference-transformer` or some other function? What stack trace do you get if you try after `toggle-debug-on-error`? My guess would be that this is some other function causing the problem.

Reply:
I have debug on error turned on, but it doesn't return a stack trace unfortunately.

Reply:
> Actually, the problem is specific to `citar-open-entry`, when accessed from `embark-act`.

This is the problem then
```elsip
(key (car key-entry))
```
This is a binding in the `while-let` inside `citar-open-entry` which expects the key-entry and not just the key.

Reply:
That's what I was thinking but that also doesn't fix it!

Time for a reboot!